lib/astutils.cpp

```diff
@@ -2419,9 +2419,15 @@ static bool isExpressionChangedAt(const F& getExprTok,
     if (depth < 0)
         return true;
     if (tok->exprId() != exprid) {
-        if (globalvar && !tok->isKeyword() && Token::Match(tok, "%name% (") && !(tok->function() && tok->function()->isAttributePure()))
-            // TODO: Is global variable really changed by function call?
-            return true;
+        if (globalvar && !tok->isKeyword() && Token::Match(tok, "%name% (") && !(tok->function() && tok->function()->isAttributePure())) {
+            const Token* exprTok = getExprTok();
+            if (!((exprTok && exprTok->variable() && exprTok->variable()->scope() && tok->function() && tok->function()->isConst() &&
+                   (exprTok->variable()->scope() == tok->function()->nestedIn || // member variable doesn't get changed within const member function
```

Comment:
We should skip this branch completely if `tok->previous()->astOperand1()->exprId()` matches the expression id.